pkgs/applications/audio/amarok/default.nix

```diff
@@ -1,23 +1,26 @@
-{ mkDerivation, fetchurl, lib
+{ stdenv, fetchurl, lib
 , extra-cmake-modules, kdoctools
 , qca-qt5, qjson, qtquickcontrols2, qtscript, qtwebengine
 , karchive, kcmutils, kconfig, kdnssd, kguiaddons, kinit, kirigami2, knewstuff, knotifyconfig, ktexteditor, kwindowsystem
-, fftw, phonon, plasma-framework, threadweaver, breeze-icons
+, fftw, phonon, plasma-framework, threadweaver, breeze-icons, wrapQtAppsHook
 , curl, ffmpeg, gdk-pixbuf, libaio, liblastfm, libmtp, loudmouth, lzo, lz4, mariadb-embedded, pcre, snappy, taglib, taglib_extras
 }:
 
-mkDerivation rec {
+stdenv.mkDerivation (finalAttrs: {
   pname = "amarok";
-  version = "2.9.71";
+  version = "3.0.0";
 
   src = fetchurl {
-    url = "mirror://kde/unstable/${pname}/${version}/${pname}-${version}.tar.xz";
-    sha256 = "0kz8wixjmy4yxq2gk11ybswryxb6alfymd3bzcar9xinscllhh3a";
+    url =
+      let name = finalAttrs.pname;
+          version = finalAttrs.version;
+      in "mirror://kde/stable/${name}/${version}/${name}-${version}.tar.xz";
+    sha256 = "sha256-FKh2eDBfrXagodrKVVpndf+mQuXrvMzs2R9JcJOZLBw=";
```

Comment:
I suggest you use the following instead:
```nix
src = fetchurl {
  url = "mirror://kde/stable/amarok/${finalAttrs.version}/amarok-${finalAttrs.version}.tar.xz";
  hash = "sha256-FKh2eDBfrXagodrKVVpndf+mQuXrvMzs2R9JcJOZLBw=";
};
```

Just because the path has `amarok` in it, doesn't mean it should be substituted for the value of `pname`. It's entirely possible that someone overrides amarok and as a consequence changes the pname - the path shouldn't change in this scenario.